Add AML-Dashboard documentation (#12)
eProsima-Private/AML-Tools: docs/rst/amldashboard/user_manual/interfaces.rst

.. include:: /rst/exports/alias.include
.. include:: /rst/exports/roles.include

.. _amldashboard_interfaces:

##########
Interfaces

> ##########

Reviewer: Please incorporate the pictures for each interface, so that it is more explicative.

Author: I will add a reference to the usage subsection where the images for each interface are explained in detail. Including the images here might be redundant since the usage section already covers this comprehensively.

The *AML Dashboard* integrates six main interfaces:

Data Collection
===============

The first screen comprises five distinct areas that enable users to swiftly create their own data sets for training an |aml| model.

> 
> 1. **Exploring Data Input**: The *AML Dashboard* collection interface is designed to help non-computer scientists easily create and understand data collection.

Reviewer: ```suggestion

1. **Exploring Data Input**: The *AML Dashboard* collection interface (Data Management tab) is designed to help non-computer scientists easily create and understand data collection.
```

Author: I believe specifying the tab is redundant.

   It features two video representations:

   - **Ground Truth Video Stream**: Shows the actual video feed.

   - **Overlaid Skeleton Video**: Displays the skeleton recognized by the hand recognition algorithm.

   These visual aids allow users to test and explore the aspects captured by the model.

2. **Collecting Data**: Users can create their own data sets to train an |aml| model.
   A drop-down menu allows the selection of one out of four musical instruments for demonstration purposes.

> 

Reviewer: Please fix this in the dashboard to add the drop-down menu, since at this moment it is not present in thisn inerface.

Author: The deliverable specifies that a drop-down menu is to be included for selecting one of four musical instruments for demonstration purposes. I believe we should remain consistent with this description. The drop-down menu is meant to refer to the instance label (I think).

> 3. **Data Set Overview**: Each collected image is displayed in a data set overview, where users can explore, move, or delete images in the training stack.

Reviewer: As of now this is done in a dataset browser. Please rewrite this section with the correct name, or adjust the name in the interface so that it matches.

```suggestion
3. **Data Set Browser**: Each collected image is displayed in a data set browser, where users can explore, move, or delete images in the training stack, as well as managing labels of the different classes.
```

   This feature is crucial for iterative training, enabling users to remove or add new labeled data to improve the |aml| algorithm as needed.

Training
========

> The training page is separate from the data collection page to educate users on the various stages of the :term:`ML` life-cycle.

Reviewer: ```suggestion
The Training tab is separated from the Data Management tab to educate users on the various stages of the :term:`ML` life-cycle.
```

> It is divided into two sections:

Reviewer: ```suggestion
It is divided in two sections:
```

> The Training page is separated from the Data Collection page to educate users on the various stages of the :term:`ML` life-cycle.

Author: I will replace this sentence with one that more clearly explains the purpose of the training page.

Reviewer: Done

1. **AML Model Training Information (Left Side)**:

>    - **Training Information**: Users can set parameters such as the number of iterations.

Reviewer: ```suggestion
   - **Training Information**: Users can set parameters such as the number of iterations or the percentage of data used for each execution.
```

>    - **Training Visualization**: Displays the current state of training.

Reviewer: ```suggestion
   - **Training Status**: Displays the current state of training.
```

2. **Comparison Model Training (Right Side)**:

   - **Model Parameters**: Users can set or alter parameters for a comparison model, such as a Multi-layer Perceptron (MLP) classifier.
     This model can be replaced with any other type of classifier for direct comparison with |aml|.

   - **Training Progress**: A loading bar shows the projected training duration.
     Below it, variations in loss and accuracy per epoch are visualized, helping experts assess the classification quality and data noise level.

Fetching
========

> Retrieve trained |aml| statistics and Models effortlessly.

Reviewer: ```suggestion
This tab's purpose is to retrieve trained |aml| statistics and Models effortlessly.
```

Author: I think it would be clearer to refer to this as an interface rather than a tab.

It includes options to search for |aml| statistics and request an |aml| model.

It is divided into the following sections:

1. **Status Display**: Shows the current status of collaborative learning, statistics, and model fetching.

2. **AML Statistics Fetcher**:

   - **Search Button**: Search for |aml| statistics.

   - **Statistics Display**: Display the fetched statistics

3. **AML Model Fetcher**:

   - **Request Button**: Request an |aml| model.

   - **Model Display**: Display the fetched model.

> Explore Models

Reviewer: Please, match the name of the interface in the documentation with the name of the real interface (Batch Prediction)

Author: I believe this page is meant to provide an understanding of the different interfaces within the dashboard and their purposes. For exact names and detailed navigation, we can refer to the usage page. This documentation focuses on explaining the functionality of each interface rather than using their literal names.

==============

The model exploration interface provides space for developers to add any desired model comparison matrix.
The Marcelle framework supports various comparison modules and allows users to create custom :term:`GUI` elements for specific calculations.

In the current setup, a confusion matrix is provided based on the trained model.
Users must actively trigger the update of the matrix, facilitating a clear before-and-after comparison.
Future work aims to extend this section with more targeted visualizations utilizing the |aml| output structure.

> Music Exploration

Reviewer: Please, make the name of this tab match the name of the real tab (Real-time Prediction)

Author: I will change the name to a more generic term since it’s meant to describe the functionality of the interface rather than focusing on specific contexts like music.

Reviewer: Done

1. **Model Selection**: Users can toggle between |aml| and *NN* models for real-time predictions.

2. **Camera Activation**: Starting the camera initiates real-time predictions from the activated model.

3. **Real-Time Predictions**: Predictions are shown in real-time.
   If both models run, the prediction confidence of the |aml| model determines the sound pattern volume.

Sound is produced by looping audio tracks of musical instruments assigned during data collection.
Users can recreate gestures to explore sound intersections using *AML Dashboard*.

Context Broker interaction
==========================

Create and update data in the Context Broker and receive solutions (inferences) from the |aml| model.

1. **Fiware Node**:

   - **Parameters**: Specify the Fiware Node parameters and Context Broker entity ID and attributes.

   - **Create Button**: Create the node using the provided parameters.

   - **Status Update**: Shows whether the node has been created successfully.

2. **Context Broker Data**:

   - **Data Upload**: Drag and drop an image or upload a file in the designated area to upload data.

   - **Post Data Button**: Send the data to the Context Broker.

>    - **Data Status**: Indicates whether the data has been successfully posted.

Reviewer: ```suggestion
   - **Data Status**: Indicates whether the data has been successfully loaded and posted.
```
